### PR TITLE
Set trajectory execution monitor params under correct namespace

### PR DESCRIPTION
**Pull request by wkentaro**
For #61 and #68

### CODE REVIEW

**Comment by davetcoleman** [comment]
+1

File: moveit_setup_assistant/templates/moveit_config_pkg_template/launch/trajectory_execution.launch.xml [hunk 1]
```diff
@@ -7,9 +7,9 @@
   <param name="moveit_manage_controllers" value="$(arg moveit_manage_controllers)"/>
 
   <!-- When determining the expected duration of a trajectory, this multiplicative factor is applied to get the allowed duration of execution -->
-  <param name="trajectory_execution/allowed_execution_duration_scaling" value="1.2"/> <!-- default 1.2 -->
+  <param name="allowed_execution_duration_scaling" value="1.2"/> <!-- default 1.2 -->
   <!-- Allow more than the expected execution time before triggering a trajectory cancel (applied after scaling) -->
-  <param name="trajectory_execution/allowed_goal_duration_margin" value="0.5"/> <!-- default 0.5 -->
+  <param name="allowed_goal_duration_margin" value="0.5"/> <!-- default 0.5 -->
```
Comment (rhaschke) [inline moveit_setup_assistant/templates/moveit_config_pkg_template/launch/trajectory_execution.launch.xml]: Why did you reverted this? IMHO it makes sense to structure the param space using the "trajectory_execution" ns.
Furthermore, the namespace is needed for dynamic reconfigure. This was the point of #68.

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp [hunk 2]
```diff
@@ -65,6 +65,7 @@ class TrajectoryExecutionManager::DynamicReconfigureImpl
   {
     owner_->enableExecutionDurationMonitoring(config.execution_duration_monitoring);
     owner_->setAllowedExecutionDurationScaling(config.allowed_execution_duration_scaling);
+    owner_->setAllowedGoalDurationMargin(config.allowed_goal_duration_margin);
   }
```
Comment (rhaschke) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp]: `setExecutionVelocityScaling()` is still missing.

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp [hunk 3]
```diff
@@ -77,6 +78,11 @@ TrajectoryExecutionManager::TrajectoryExecutionManager(const robot_model::RobotM
   if (!node_handle_.getParam("moveit_manage_controllers", manage_controllers_))
     manage_controllers_ = false;
 
+  ROS_WARN_NAMED("trajectory_execution_manager",
+                 "Deprecation warning: the namespace for params ~allowed_execution_duration_scaling and "
+                 "~allowed_goal_duration_margin are changed under ns `~/trajectory_execution`."
+                 "Parameter reading from that under old ns will be deprecated.");
+
   if (!node_handle_.getParam("allowed_execution_duration_scaling", allowed_execution_duration_scaling_))
```
Comment (rhaschke) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp]: The warning should only be issues when the parameter was found in the _wrong_ namespace.

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp (hunk 2 above)
Comment (wkentaro) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp]: I made it in another PR #73 .

File: moveit_setup_assistant/templates/moveit_config_pkg_template/launch/trajectory_execution.launch.xml (hunk 1 above)
Comment (rhaschke) [inline moveit_setup_assistant/templates/moveit_config_pkg_template/launch/trajectory_execution.launch.xml]: Sorry, obviously I didn't look at the most recent code...

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp (hunk 2 above)
Comment (rhaschke) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp]: I see.

**Comment by wkentaro** [comment]
@rhaschke Thanks for the review. Rebased, and fixed based on the comments.

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp (hunk 3 above)
Comment (wkentaro) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp]: That's reasonable. I will fix it.

Comment (rhaschke) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp]: I suggest a shorter warning, e.g.:
const char* DEPRECATION_WARNING="Deprecation warning: parameter %s should be moved into namespace 'trajectory_execution'. Adjust file trajectory_execution.launch.xml."
ROS_WARN_NAMED("trajectory_execution_manager", DEPRECATION_WARNING, "allowed_goal_duration_margin");

Sorry for being too slow for you. I was interrupted.

**Comment by rhaschke** [comment]
Please consider https://github.com/ros-planning/moveit/pull/72#discussion_r75270492. Otherwise looks good to me.

**Comment by wkentaro** [comment]
Thanks. Fixed.

**Comment by rhaschke** [comment]
Merged in kinetic first (988ad5b7) and cherry-picked to jade and indigo.
@wkentaro Thanks for your effort.

**Comment by wkentaro** [comment]
Thanks.

**Pull request closed without merging**